### PR TITLE
feat: integrate simplehash

### PR DESCRIPTION
**Pull request by hanszhoou**
## Description
https://mask.atlassian.net/browse/MF-3634

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

**Review by guanbinrui — DISMISSED** [review]

File: packages/web3-state/src/Hub/NonFungibleClient.ts [hunk 1]
```diff
@@ -89,13 +89,26 @@ export class HubStateNonFungibleClient<ChainId, SchemaType> extends HubStateBase
     }
 
     async getNonFungibleAssetsByCollection(
-        address: string,
+        id: string,
```
Comment (guanbinrui) [inline packages/web3-state/src/Hub/NonFungibleClient.ts]:
```suggestion
        address: string,
```

File: packages/web3-state/src/Hub/NonFungibleClient.ts [hunk 2]
```diff
@@ -89,13 +89,26 @@ export class HubStateNonFungibleClient<ChainId, SchemaType> extends HubStateBase
...
+        const options = this.getOptions(initial)
+        const providers = this.getProviders(initial)
+        return attemptUntil(
+            providers.map((x) => () => x.getAssetsByCollection?.(id, options)),
+            createPageable(EMPTY_LIST, createIndicator(options.indicator)),
+        )
+    }
+
+    async getNonFungibleAssetsByCollectionAndOwner(
+        collectionId: string,
```
Comment (guanbinrui) [inline packages/web3-state/src/Hub/NonFungibleClient.ts:104]:
```suggestion
        address: string,
```

File: packages/web3-shared/base/src/specs/index.ts [hunk 3]
```diff
@@ -1314,7 +1316,13 @@ export interface HubFungible<ChainId, SchemaType, GasOption, Web3HubOptions = Hu
 export interface HubNonFungible<ChainId, SchemaType, GasOption, Web3HubOptions = HubOptions<ChainId>> {
     /** Get non-fungible assets of the given collection. */
     getNonFungibleAssetsByCollection?: (
-        address: string,
+        id: string,
```
Comment (guanbinrui) [inline packages/web3-shared/base/src/specs/index.ts]:
```suggestion
        address: string,
```

File: packages/web3-shared/base/src/specs/index.ts [hunk 4]
```diff
@@ -1314,7 +1316,13 @@ export interface HubFungible<ChainId, SchemaType, GasOption, Web3HubOptions = Hu
...
+        initial?: Web3HubOptions,
+    ) => Promise<Pageable<NonFungibleAsset<ChainId, SchemaType>>>
+    /** Get non-fungible assets of the given collection and owner. */
+    getNonFungibleAssetsByCollectionAndOwner?: (
+        collectionId: string,
```
Comment (guanbinrui) [inline packages/web3-shared/base/src/specs/index.ts:1324]:
```suggestion
        address: string,
```

File: packages/web3-providers/src/types/NonFungibleToken.ts [hunk 5]
```diff
@@ -99,6 +99,12 @@ export namespace NonFungibleTokenAPI {
             address: string,
             options?: HubOptions<ChainId>,
         ) => Promise<Pageable<NonFungibleAsset<ChainId, SchemaType>>>
+        /** Get non-fungible assets of the given collection and owner. */
+        getAssetsByCollectionAndOwner?: (
+            collectionId: string,
```
Comment (guanbinrui) [inline packages/web3-providers/src/types/NonFungibleToken.ts:104]:
```suggestion
            address: string,
```

File: packages/web3-providers/src/NFTScan/apis/NonFungibleTokenAPI_EVM.ts [hunk 6]
```diff
@@ -51,13 +51,13 @@ export class NFTScanNonFungibleTokenAPI_EVM implements NonFungibleTokenAPI.Provi
     }
 
     async getAssetsByCollection(
-        address: string,
+        id: string,
```
Comment (guanbinrui) [inline packages/web3-providers/src/NFTScan/apis/NonFungibleTokenAPI_EVM.ts]:
```suggestion
        address: string,
```

File: packages/web3-hooks/base/src/useNonFungibleAssetsByCollection.ts [hunk 7]
```diff
@@ -7,7 +7,7 @@ import { useWeb3Hub } from './useWeb3Hub.js'
 export function useNonFungibleAssetsByCollection<
     S extends 'all' | void = void,
     T extends NetworkPluginID = NetworkPluginID,
->(address?: string, pluginID?: T, options?: Web3Helper.Web3HubOptionsScope<S, T>) {
+>(id?: string, pluginID?: T, options?: Web3Helper.Web3HubOptionsScope<S, T>) {
```
Comment (guanbinrui) [inline packages/web3-hooks/base/src/useNonFungibleAssetsByCollection.ts]:
```suggestion
>(address?: string, pluginID?: T, options?: Web3Helper.Web3HubOptionsScope<S, T>) {
```

File: packages/web3-hooks/base/src/useNonFungibleAssetsByCollectionAndOwner.ts [hunk 8]
```diff
@@ -0,0 +1,82 @@
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import { pageableToIterator } from '@masknet/web3-shared-base'
+import { EMPTY_LIST, type NetworkPluginID } from '@masknet/shared-base'
+import type { Web3Helper } from '@masknet/web3-helpers'
+import { useWeb3Hub } from './useWeb3Hub.js'
+
+export function useNonFungibleAssetsByCollectionAndOwner<
+    S extends 'all' | void = void,
+    T extends NetworkPluginID = NetworkPluginID,
+>(collectionId?: string, owner?: string, pluginID?: T, options?: Web3Helper.Web3HubOptionsScope<S, T>) {
```
Comment (guanbinrui) [inline packages/web3-hooks/base/src/useNonFungibleAssetsByCollectionAndOwner.ts:10]:
```suggestion
>(address?: string, owner?: string, pluginID?: T, options?: Web3Helper.Web3HubOptionsScope<S, T>) {
```

File: packages/web3-hooks/base/src/useNonFungibleAssetsByCollectionAndOwner.ts [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+                    }
+                }
+            }
+        } catch (error_) {
+            // Uncontrolled error
+            setError(error_ as string)
+            setDone(true)
+        }
+        toggleLoading(false)
+        setAssets((pred) => [...pred, ...batchResult].sort((a, b) => Number(a.tokenId) - Number(b.tokenId)))
```
Comment (guanbinrui) [inline packages/web3-hooks/base/src/useNonFungibleAssetsByCollectionAndOwner.ts]: Token ID could be a no-numeric string.

File: packages/web3-providers/src/SimpleHash/index.ts [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+            NonFungibleAsset<ChainId, SchemaType>
+        >
+
+        const x = createPageable(
+            assets,
+            createIndicator(indicator),
+            response.next_cursor ? createNextIndicator(indicator, response.next_cursor) : undefined,
+        )
+
+        return x
```
Comment (guanbinrui) [inline packages/web3-providers/src/SimpleHash/index.ts]:
```suggestion
        return createPageable(
            assets,
            createIndicator(indicator),
            response.next_cursor ? createNextIndicator(indicator, response.next_cursor) : undefined,
        )
```

File: packages/web3-hooks/base/src/useNonFungibleAssetsByCollectionAndOwner.ts (hunk 9 above)
Comment (hanszhoou) [reply packages/web3-hooks/base/src/useNonFungibleAssetsByCollectionAndOwner.ts]: I think it's okay, no error.

**Pull request merged**